jena-core/src/main/java/org/apache/jena/util/JenaXMLInput.java

```diff
@@ -72,16 +72,24 @@ public static XMLReader createXMLReader() throws ParserConfigurationException, S
      * Initialize an XMLInputFactory to jena settings.
      */
     public static void initXMLInputFactory(XMLInputFactory xf) {
+        // This disables DTDs entirely for the factory.
+        // All DTDs are silently ignored; takes precedence over ACCESS_EXTERNAL_DTD
+    	setXMLInputFactoryProperty(xf, XMLInputFactory.SUPPORT_DTD, Boolean.FALSE);
+
+        // Disable external DTDs (files and HTTP) - errors unless SUPPORT_DTD is false.
+        setXMLInputFactoryProperty(xf, XMLConstants.ACCESS_EXTERNAL_DTD, "");
+
+        // disable external entities (silently ignore)
+        setXMLInputFactoryProperty(xf, XMLInputFactory.IS_SUPPORTING_EXTERNAL_ENTITIES, Boolean.FALSE);
+    }
+
+    /**
+     * Catch any {@link IllegalArgumentException}, log it, and continue.
+     */
+    private static void setXMLInputFactoryProperty(XMLInputFactory xf, String name, Object value) {
         try {
-            // This disables DTDs entirely for the factory.
-            // All DTDs are silently ignored; takes precedence over ACCESS_EXTERNAL_DTD
-            xf.setProperty(XMLInputFactory.SUPPORT_DTD, false);
-
-            // Disable external DTDs (files and HTTP) - errors unless SUPPORT_DTD is false.
-            xf.setProperty(XMLConstants.ACCESS_EXTERNAL_DTD, "");
-            // disable external entities (silently ignore)
-            xf.setProperty("javax.xml.stream.isSupportingExternalEntities", false);
-        } catch(IllegalArgumentException ex){
+            xf.setProperty(name, value);
+        } catch(IllegalArgumentException ex) {
             Log.error(JenaXMLInput.class, "Problem setting StAX property", ex);
```

Comment:
Assuming this is now to catch specific case, the stacktrace isn't necessary (reduces the log output), if we have the exception message and the specific property `name` included instead.